### PR TITLE
fix(graphql): remove non-null constraints for active recs action

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] Signed the last commit: `git commit --amend --signoff`
_______________________________________________

Related to https://github.com/cryostatio/cryostat-web/pull/906

## Description of the change:

Remove non-null constraints for active recording actions. This allows data to be returned when an error occurs.

## Motivation for the change:

See https://github.com/cryostatio/cryostat-web/pull/906#discussion_r1137825159

## How to manually test:

Try on JDP group with some targets that cannot start/stop/delete recordings. Check the response for the `data` field.

### Start Recording

```graphql
query StartRecordingForGroup() {
  environmentNodes(filter: { name: "JDP" }) {
    name
    descendantTargets {
      name
      doStartRecording(recording: {
        name: "foo",
        template: "Continuous",
        templateType: "TARGET",
        duration: 0,
        restart: true,
        metadata: {
          labels: [
            {
              key: "cryostat.io.topology-group",
              value: "JDP"
            }
          ]
        },
        }) {
            name
            state
      }
    }
  }
}
```

### Stop Recording

```graphql
query StopRecordingForGroup (){
    environmentNodes(filter: { name: "JDP" }) {
      name
      descendantTargets {
        name
        recordings {
            active(filter: { name: "foo" }) {
                data {
                    doStop {
                      name
                      state
                    }
                }
            }
        }
      }
    }
}
```

### Delete Recording

```graphql
query DeleteRecordingForGroup (){
    environmentNodes(filter: { name: "JDP" }) {
      name
      descendantTargets {
        name
        recordings {
            active(filter: { name: "foo" }) {
                data {
                    doDelete {
                      name
                      state
                    }
                }
            }
        }
      }
    }
}
```

### CODE REVIEW

**Comment by tthvo** [comment]
Error response:

```json
{
   "errors":[
      {
         "message":"Exception while fetching data (/environmentNodes[0]/descendantTargets[0]/doStartRecording) : org.openjdk.jmc.rjmx.ConnectionException caused by java.lang.SecurityException: Authentication failed! Invalid username or password",
         "locations":[
            {
               "line":1,
               "column":161
            }
         ],
         "path":[
            "environmentNodes",
            0,
            "descendantTargets",
            0,
            "doStartRecording"
         ],
         "extensions":{
            "classification":"DataFetchingException"
         }
      },
      {
         "message":"Exception while fetching data (/environmentNodes[0]/descendantTargets[2]/doStartRecording) : org.openjdk.jmc.rjmx.ConnectionException caused by java.lang.SecurityException: Authentication failed! Invalid username or password",
         "locations":[
            {
               "line":1,
               "column":161
            }
         ],
         "path":[
            "environmentNodes",
            0,
            "descendantTargets",
            2,
            "doStartRecording"
         ],
         "extensions":{
            "classification":"DataFetchingException"
         }
      },
      {
         "message":"Exception while fetching data (/environmentNodes[0]/descendantTargets[3]/doStartRecording) : org.openjdk.jmc.rjmx.ConnectionException caused by java.io.IOException: Failed to retrieve RMIServer stub: javax.naming.CommunicationException [Root exception is java.rmi.ConnectIOException: non-JRMP server at remote endpoint]",
         "locations":[
            {
               "line":1,
               "column":161
            }
         ],
         "path":[
            "environmentNodes",
            0,
            "descendantTargets",
            3,
            "doStartRecording"
         ],
         "extensions":{
            "classification":"DataFetchingException"
         }
      }
   ],
   "data":{
      "environmentNodes":[
         {
            "name":"JDP",
            "descendantTargets":[
               {
                  "name":"service:jmx:rmi:///jndi/rmi://cryostat:9091/jmxrmi",
                  "doStartRecording":null
               },
               {
                  "name":"service:jmx:rmi:///jndi/rmi://cryostat:9093/jmxrmi",
                  "doStartRecording":{
                     "name":"cryostat_topology_action",
                     "state":"RUNNING"
                  }
               },
               {
                  "name":"service:jmx:rmi:///jndi/rmi://cryostat:9094/jmxrmi",
                  "doStartRecording":null
               },
               {
                  "name":"service:jmx:rmi:///jndi/rmi://cryostat:9095/jmxrmi",
                  "doStartRecording":null
               }
            ]
         }
      ]
   }
}
```

File: src/main/resources/types.graphqls [hunk 1]
```diff
@@ -67,15 +67,15 @@ type Annotations {
 
 type TargetNode implements Node {
     target: ServiceRef!
-    recordings: Recordings!
+    recordings: Recordings
     mbeanMetrics: MBeanMetrics!
```
Comment (andrewazores) [inline src/main/resources/types.graphqls:71]: `mbeanMetrics` should also allow null responses for the same reasons as the mutations, I think.

File: src/main/resources/types.graphqls [hunk 2]
```diff
@@ -67,15 +67,15 @@ type Annotations {
...
 
     id: Int!
     name: String!
     nodeType: NodeType!
     labels: Object!
 
-    doStartRecording(recording: RecordingSettings!): ActiveRecording!
+    doStartRecording(recording: RecordingSettings!): ActiveRecording
     doSnapshot: ActiveRecording!
```
Comment (andrewazores) [inline src/main/resources/types.graphqls:79]: `doSnapshot` and `doPutMetadata` below as well

**Review by andrewazores — APPROVED** [review]

**Comment by andrewazores** [comment]
@tthvo rebase + (re)sign commit please

**Review by andrewazores — APPROVED** [review]

**Pull request merged**